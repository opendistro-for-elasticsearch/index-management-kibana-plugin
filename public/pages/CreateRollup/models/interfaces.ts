/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { ManagedCatIndex } from "../../../../server/models/interfaces";
import { FieldItem } from "../../../../models/interfaces";

export interface MappingsItem {
  properties?: {};
  type?: string;
}

export interface DimensionItem {
  sequence: number;
  field: FieldItem;
  aggregationMethod: string;
  interval?: number;
}

export interface MetricItem {
  source_field: FieldItem;
  all: boolean;
  min: boolean;
  max: boolean;
  sum: boolean;
  avg: boolean;
  value_count: boolean;
}

//From backend
export interface RollupDimensionItem {
  date_histogram?: {
    fixed_interval?: string;
    calendar_interval?: string;
    source_field: string;
    target_field: string;
    timezone: string;
  };
  terms?: {
    source_field: string;
    target_field: string;
  };
  histogram?: {
    source_field: string;
    target_field: string;
    interval: number;
  };
}

export interface RollupMetricItem {
  source_field: string;
  metrics: [
    {
      min?: Object;
      max?: Object;
      sum?: Object;
      avg?: Object;
      value_count?: Object;
    }
  ];
}
