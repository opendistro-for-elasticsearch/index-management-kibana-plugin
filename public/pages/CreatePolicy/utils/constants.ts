/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const DEFAULT_POLICY = JSON.stringify(
  {
    policy: {
      description: "A simple default policy that changes the replica count between hot and cold states.",
      default_state: "hot",
      states: [
        {
          name: "hot",
          actions: [{ replica_count: { number_of_replicas: 5 } }],
          transitions: [{ state_name: "cold", conditions: { min_index_age: "30d" } }],
        },
        {
          name: "cold",
          actions: [{ replica_count: { number_of_replicas: 2 } }],
          transitions: [],
        },
      ],
    },
  },
  null,
  4
);
