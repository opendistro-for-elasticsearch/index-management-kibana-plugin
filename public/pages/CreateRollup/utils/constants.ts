/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

//Added last_updated_time and start_time to avoid error, but they should be system generated.
//Added a delay as 0, but this should be optional, so null should be fine.
export const EMPTY_ROLLUP = JSON.stringify({
  rollup: {
    continuous: false,
    description: "",
    dimensions: [
      {
        date_histogram: {
          source_field: "",
          fixed_interval: "2ms",
          timezone: "Africa/Abidjan",
        },
      },
    ],
    enabled: false,
    metrics: [],
    page_size: 1000,
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const FixedTimeunitOptions = [
  { value: "ms", text: "Millisecond(s)" },
  { value: "s", text: "Second(s)" },
  { value: "m", text: "Minute(s)" },
  { value: "h", text: "Hour(s)" },
  { value: "d", text: "Day(s)" },
];

export const DelayTimeunitOptions = [
  { value: "SECONDS", text: "Second(s)" },
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const CalendarTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const AddFieldsColumns = [
  {
    field: "label",
    name: "Field name",
    sortable: true,
  },
  {
    field: "type",
    name: "Field type",
    sortable: true,
    render: (type: string | undefined) => (type == null || type == undefined ? "-" : type),
  },
];
