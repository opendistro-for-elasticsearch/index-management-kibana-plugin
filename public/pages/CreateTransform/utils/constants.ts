/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

export const EMPTY_TRANSFORM = JSON.stringify({
  transform: {
    description: "",
    enabled: true,
    page_size: 1000,
    roles: [],
    schedule: {
      interval: {
        start_time: 234802,
        period: 1,
        unit: "MINUTES",
      },
    },
    source_index: "",
    target_index: "",
  },
});

export const FixedTimeunitOptions = [
  { value: "ms", text: "Millisecond(s)" },
  { value: "s", text: "Second(s)" },
  { value: "m", text: "Minute(s)" },
  { value: "h", text: "Hour(s)" },
  { value: "d", text: "Day(s)" },
];

export const DelayTimeunitOptions = [
  { value: "SECONDS", text: "Second(s)" },
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const CalendarTimeunitOptions = [
  { value: "m", text: "Minute" },
  { value: "h", text: "Hour" },
  { value: "d", text: "Day" },
  { value: "w", text: "Week" },
  { value: "M", text: "Month" },
  { value: "q", text: "Quarter" },
  { value: "y", text: "Year" },
];

export const ScheduleIntervalTimeunitOptions = [
  { value: "MINUTES", text: "Minute(s)" },
  { value: "HOURS", text: "Hour(s)" },
  { value: "DAYS", text: "Day(s)" },
];

export const AddFieldsColumns = [
  {
    field: "label",
    name: "Field name",
    sortable: true,
  },
  {
    field: "type",
    name: "Field type",
    sortable: true,
    render: (type: string | undefined) => (type == null || type == undefined ? "-" : type),
  },
];

export const ExecutionFrequencyDefinitionOptions = [
  { value: "fixed", text: "Define by fixed interval" },
  { value: "cron", text: "Define by cron expression" },
];

export const DefaultSampleDataSize = 50;
