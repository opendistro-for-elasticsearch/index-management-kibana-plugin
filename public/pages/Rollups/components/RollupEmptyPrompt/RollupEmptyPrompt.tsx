/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import { EuiButton, EuiEmptyPrompt, EuiText } from "@elastic/eui";
import { PLUGIN_NAME, ROUTES } from "../../../../utils/constants";

export const TEXT = {
  RESET_FILTERS: "There are no rollup jobs matching your applied filters. Reset your filters to view your rollup jobs.",
  NO_ROLLUPS:
    "Rollup jobs help you conserve storage space for historical time series data while preserving the specific information you need.",
  LOADING: "Loading rollup jobs...",
};

const getMessagePrompt = ({ filterIsApplied, loading }: RollupEmptyPromptProps) => {
  if (loading) return TEXT.LOADING;
  if (filterIsApplied) return TEXT.RESET_FILTERS;
  return TEXT.NO_ROLLUPS;
};

const getActions: React.SFC<RollupEmptyPromptProps> = ({ filterIsApplied, loading, resetFilters }) => {
  if (loading) {
    return null;
  }
  if (filterIsApplied) {
    return (
      <EuiButton fill onClick={resetFilters} data-test-subj="rollupEmptyPromptRestFilters">
        Reset Filters
      </EuiButton>
    );
  }

  return (
    <EuiButton href={`${PLUGIN_NAME}#${ROUTES.CREATE_ROLLUP}`} data-test-subj="emptyPromptCreateRollupButton">
      Create rollup
    </EuiButton>
  );
};

interface RollupEmptyPromptProps {
  filterIsApplied: boolean;
  loading: boolean;
  resetFilters: () => void;
}

const RollupEmptyPrompt: React.SFC<RollupEmptyPromptProps> = (props) => (
  <EuiEmptyPrompt
    style={{ maxWidth: "45em" }}
    body={
      <EuiText>
        <p>{getMessagePrompt(props)}</p>
      </EuiText>
    }
    actions={getActions(props)}
  />
);

export default RollupEmptyPrompt;
