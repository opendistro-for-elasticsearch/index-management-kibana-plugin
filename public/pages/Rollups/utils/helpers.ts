/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import queryString from "query-string";
// @ts-ignore
import moment from "moment";
import { DEFAULT_QUERY_PARAMS } from "./constants";
import { RollupQueryParams } from "../models/interfaces";

export function getURLQueryParams(location: { search: string }): RollupQueryParams {
  const { from, size, search, sortField, sortDirection } = queryString.parse(location.search);

  return <RollupQueryParams>{
    // @ts-ignores
    from: isNaN(parseInt(from, 10)) ? DEFAULT_QUERY_PARAMS.from : parseInt(from, 10),
    // @ts-ignore
    size: isNaN(parseInt(size, 10)) ? DEFAULT_QUERY_PARAMS.size : parseInt(size, 10),
    search: typeof search !== "string" ? DEFAULT_QUERY_PARAMS.search : search,
    sortField: typeof sortField !== "string" ? DEFAULT_QUERY_PARAMS.sortField : sortField,
    sortDirection: typeof sortDirection !== "string" ? DEFAULT_QUERY_PARAMS.sortDirection : sortDirection,
  };
}

export const renderTime = (time: number): string => {
  const momentTime = moment(time).local();
  if (time && momentTime.isValid()) return momentTime.format("MM/DD/YY h:mm a");
  return "-";
};

export const renderEnabled = (isEnabled: boolean): string => {
  return isEnabled ? "Enabled" : "Disabled";
};

export const renderContinuous = (continuous: boolean): string => {
  return continuous ? "Yes" : "No";
};
