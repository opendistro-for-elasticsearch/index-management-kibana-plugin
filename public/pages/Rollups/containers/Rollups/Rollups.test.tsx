/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import chrome from "ui/chrome";
import { toastNotifications } from "ui/notify";
import { render, wait } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter as Router } from "react-router";
import { browserServicesMock } from "../../../../../test/mocks";
import { ServicesConsumer, ServicesContext } from "../../../../services";
import { BrowserServices } from "../../../../models/interfaces";
import { ModalProvider, ModalRoot } from "../../../../components/Modal";
import { Redirect, Route, RouteComponentProps, Switch } from "react-router-dom";
import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
import Rollups from "./Rollups";
import { TEXT } from "../../components/RollupEmptyPrompt/RollupEmptyPrompt";
import { testRollup, testRollup2 } from "../../../CreateRollup/utils/constants";

function renderRollupsWithRouter() {
  return {
    ...render(
      <Router>
        <ServicesContext.Provider value={browserServicesMock}>
          <ServicesConsumer>
            {(services: BrowserServices | null) =>
              services && (
                <ModalProvider>
                  <ModalRoot services={services} />
                  <Switch>
                    <Route
                      path={ROUTES.ROLLUPS}
                      render={(props: RouteComponentProps) => (
                        <div style={{ padding: "25px 25px" }}>
                          <Rollups {...props} rollupService={services.rollupService} />
                        </div>
                      )}
                    />
                    <Route path={ROUTES.CREATE_ROLLUP} render={(props) => <div>Testing create rollup</div>} />
                    <Route path={ROUTES.EDIT_ROLLUP} render={(props) => <div>Testing edit rollup: {props.location.search}</div>} />
                    <Route path={ROUTES.ROLLUP_DETAILS} render={(props) => <div>Testing rollup details: {props.location.search}</div>} />
                    <Redirect from="/" to={ROUTES.ROLLUPS} />
                  </Switch>
                </ModalProvider>
              )
            }
          </ServicesConsumer>
        </ServicesContext.Provider>
      </Router>
    ),
  };
}

describe("<Rollups /> spec", () => {
  it("renders the component", async () => {
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups: [], totalRollups: 0 },
    });
    const { container } = renderRollupsWithRouter();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("shows LOADING on mount", async () => {
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups: [], totalRollups: 0 },
    });
    const { getByText } = renderRollupsWithRouter();

    getByText(TEXT.LOADING);
  });

  it("sets breadcrumbs when mounting", async () => {
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups: [], totalRollups: 0 },
    });
    renderRollupsWithRouter();

    expect(chrome.breadcrumbs.set).toHaveBeenCalledTimes(1);
    expect(chrome.breadcrumbs.set).toHaveBeenCalledWith([BREADCRUMBS.INDEX_MANAGEMENT, BREADCRUMBS.ROLLUPS]);
  });

  it("loads rollups", async () => {
    const rollups = [testRollup];
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups, totalRollups: 1 },
    });
    const { getByText } = renderRollupsWithRouter();
    await wait();

    await wait(() => getByText(testRollup._id));
  });

  it("adds error toaster when get rollups has error", async () => {
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({ ok: false, error: "some error" });
    renderRollupsWithRouter();

    await wait();

    expect(toastNotifications.addDanger).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("some error");
  });

  it("adds error toaster when get rollups throws error", async () => {
    browserServicesMock.rollupService.getRollups = jest.fn().mockRejectedValue(new Error("rejected error"));
    renderRollupsWithRouter();

    await wait();

    expect(toastNotifications.addDanger).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("rejected error");
  });

  it("adds error toaster when explain API has error", async () => {
    const rollups = [testRollup];
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups, totalRollups: 1 },
    });
    browserServicesMock.rollupService.explainRollup = jest.fn().mockResolvedValue({
      ok: false,
      error: "some explain API error",
    });
    renderRollupsWithRouter();

    await wait();

    expect(toastNotifications.addDanger).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("some explain API error");
  });

  it("adds error toaster when explain throws error", async () => {
    const rollups = [testRollup];
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups, totalRollups: 1 },
    });
    browserServicesMock.rollupService.explainRollup = jest.fn().mockRejectedValue(new Error("explain API rejected error"));
    renderRollupsWithRouter();

    await wait();

    expect(toastNotifications.addDanger).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("explain API rejected error");
  });

  it("can route to create rollup", async () => {
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups: [], totalrollups: 0 },
    });
    const { getByText, getByTestId } = renderRollupsWithRouter();

    await wait();

    userEvent.click(getByTestId("createRollupButton"));

    await wait(() => getByText("Testing create rollup"));
  });

  it("can route to edit rollup", async () => {
    const rollups = [testRollup];
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups, totalrollups: 1 },
    });
    const { getByText, getByTestId } = renderRollupsWithRouter();

    await wait(() => getByText(testRollup._id));

    userEvent.click(getByTestId(`checkboxSelectRow-${testRollup._id}`));

    userEvent.click(getByTestId("actionButton"));

    await wait(() => getByTestId("editButton"));

    userEvent.click(getByTestId("editButton"));

    await wait(() => getByText(`Testing edit rollup: ?id=${testRollup._id}`));
  });

  //Cannot edit multiple jobs (P1)

  it("can view details of a rollup job", async () => {
    const rollups = [testRollup];
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups, totalRollups: 1 },
    });
    const { getByText } = renderRollupsWithRouter();

    await wait();
    await wait(() => getByText(testRollup._id));

    userEvent.click(getByText(testRollup._id));

    await wait(() => getByText(`Testing rollup details: ?id=${testRollup._id}`));
  });

  it("can enable a rollup job", async () => {
    const rollups = [testRollup];
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups, totalRollups: 1 },
    });
    browserServicesMock.rollupService.startRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: true,
    });
    const { getByText, getByTestId } = renderRollupsWithRouter();

    await wait(() => getByText(testRollup._id));

    expect(getByText("Enable")).toBeDisabled();

    userEvent.click(getByTestId(`checkboxSelectRow-${testRollup._id}`));

    expect(getByText("Enable")).toBeEnabled();

    userEvent.click(getByText("Enable"));

    await wait();

    expect(browserServicesMock.rollupService.startRollup).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addSuccess).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addSuccess).toHaveBeenCalledWith(`${testRollup._id} is enabled`);
  });

  //Cannot enable a job

  it("can disable a rollup job", async () => {
    const rollups = [testRollup];
    browserServicesMock.rollupService.getRollups = jest.fn().mockResolvedValue({
      ok: true,
      response: { rollups, totalRollups: 1 },
    });
    browserServicesMock.rollupService.stopRollup = jest.fn().mockResolvedValue({
      ok: true,
      response: true,
    });

    const { getByText, getByTestId } = renderRollupsWithRouter();

    await wait(() => getByText(testRollup._id));

    expect(getByText("Disable")).toBeDisabled();

    userEvent.click(getByTestId(`checkboxSelectRow-${testRollup._id}`));

    expect(getByText("Disable")).toBeEnabled();

    userEvent.click(getByText("Disable"));

    await wait();

    expect(browserServicesMock.rollupService.stopRollup).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addSuccess).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addSuccess).toHaveBeenCalledWith(`${testRollup._id} is disabled`);
  });
});
