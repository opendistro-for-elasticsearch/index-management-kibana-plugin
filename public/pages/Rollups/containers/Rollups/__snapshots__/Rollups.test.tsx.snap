// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rollups /> spec renders the component 1`] = `
<div
  style="padding: 25px 25px;"
>
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--medium"
        >
          Rollup jobs (0)
        </h3>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              class="euiButton euiButton--primary euiButton-isDisabled"
              data-test-subj="disableButton"
              disabled=""
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Disable
                </span>
              </span>
            </button>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              class="euiButton euiButton--primary euiButton-isDisabled"
              data-test-subj="enableButton"
              disabled=""
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Enable
                </span>
              </span>
            </button>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <div
              class="euiPopover euiPopover--anchorDownLeft"
              data-test-subj="actionPopover"
              id="action"
            >
              <div
                class="euiPopover__anchor"
              >
                <button
                  class="euiButton euiButton--primary euiButton-isDisabled"
                  data-test-subj="actionButton"
                  disabled=""
                  type="button"
                >
                  <span
                    class="euiButtonContent euiButtonContent--iconRight euiButton__content"
                  >
                    EuiIconMock
                    <span
                      class="euiButton__text"
                    >
                      Actions
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <button
              class="euiButton euiButton--primary euiButton--fill"
              data-test-subj="createRollupButton"
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Create rollup job
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      style="padding: initial;"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        style="padding: 0px 5px;"
      >
        <div
          class="euiFlexItem"
        >
          <div
            class="euiFormControlLayout euiFormControlLayout--fullWidth"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                class="euiFieldSearch euiFieldSearch--fullWidth euiFieldSearch-isClearable"
                placeholder="Search"
                type="search"
                value=""
              />
              <div
                class="euiFormControlLayoutIcons"
              >
                <span
                  class="euiFormControlLayoutCustomIcon"
                >
                  EuiIconMock
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
      />
      <div
        class="euiBasicTable"
      >
        <div>
          <div
            class="euiTableHeaderMobile"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsBaseline euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow"
            >
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  class="euiCheckbox"
                >
                  <input
                    aria-label="Select all rows"
                    class="euiCheckbox__input"
                    disabled=""
                    id="_selection_column-checkbox_some_html_id"
                    type="checkbox"
                  />
                  <div
                    class="euiCheckbox__square"
                  />
                  <label
                    class="euiCheckbox__label"
                    for="_selection_column-checkbox_some_html_id"
                  >
                    Select all rows
                  </label>
                </div>
              </div>
              <div
                class="euiFlexItem euiFlexItem--flexGrowZero"
              >
                <div
                  class="euiTableSortMobile"
                >
                  <div
                    class="euiPopover euiPopover--anchorDownRight"
                  >
                    <div
                      class="euiPopover__anchor"
                    >
                      <button
                        class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--xSmall euiButtonEmpty--flushRight"
                        type="button"
                      >
                        <span
                          class="euiButtonContent euiButtonContent--iconRight euiButtonEmpty__content"
                        >
                          EuiIconMock
                          <span
                            class="euiButtonEmpty__text"
                          >
                            Sorting
                          </span>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <table
            class="euiTable euiTable--responsive euiTable--auto"
            id="some_html_id"
            tabindex="-1"
          >
            <caption
              class="euiScreenReaderOnly euiTableCaption"
            />
            <thead>
              <tr>
                <th
                  class="euiTableHeaderCellCheckbox"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <div
                      class="euiCheckbox euiCheckbox--inList euiCheckbox--noLabel"
                    >
                      <input
                        aria-label="Select all rows"
                        class="euiCheckbox__input"
                        data-test-subj="checkboxSelectAll"
                        disabled=""
                        id="_selection_column-checkbox_some_html_id"
                        type="checkbox"
                      />
                      <div
                        class="euiCheckbox__square"
                      />
                    </div>
                  </div>
                </th>
                <th
                  aria-live="polite"
                  aria-sort="descending"
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell__id_0"
                  role="columnheader"
                  scope="col"
                >
                  <button
                    class="euiTableHeaderButton euiTableHeaderButton-isSorted"
                    data-test-subj="tableHeaderSortButton"
                    type="button"
                  >
                    <span
                      class="euiTableCellContent"
                    >
                      <span
                        class="euiTableCellContent__text"
                        title="Name; Sorted in descending order"
                      >
                        Name
                      </span>
                      EuiIconMock
                      <span
                        class="euiScreenReaderOnly"
                      >
                        Click to sort in ascending order
                      </span>
                    </span>
                  </button>
                </th>
                <th
                  aria-live="polite"
                  aria-sort="none"
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_rollup.source_index_1"
                  role="columnheader"
                  scope="col"
                >
                  <button
                    class="euiTableHeaderButton"
                    data-test-subj="tableHeaderSortButton"
                    type="button"
                  >
                    <span
                      class="euiTableCellContent"
                    >
                      <span
                        class="euiTableCellContent__text"
                        title="Source index"
                      >
                        Source index
                      </span>
                      <span
                        class="euiScreenReaderOnly"
                      >
                        Click to sort in ascending order
                      </span>
                    </span>
                  </button>
                </th>
                <th
                  aria-live="polite"
                  aria-sort="none"
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_rollup.target_index_2"
                  role="columnheader"
                  scope="col"
                >
                  <button
                    class="euiTableHeaderButton"
                    data-test-subj="tableHeaderSortButton"
                    type="button"
                  >
                    <span
                      class="euiTableCellContent"
                    >
                      <span
                        class="euiTableCellContent__text"
                        title="Target index"
                      >
                        Target index
                      </span>
                      <span
                        class="euiScreenReaderOnly"
                      >
                        Click to sort in ascending order
                      </span>
                    </span>
                  </button>
                </th>
                <th
                  aria-live="polite"
                  aria-sort="none"
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_rollup.rollup.enabled_3"
                  role="columnheader"
                  scope="col"
                >
                  <button
                    class="euiTableHeaderButton"
                    data-test-subj="tableHeaderSortButton"
                    type="button"
                  >
                    <span
                      class="euiTableCellContent"
                    >
                      <span
                        class="euiTableCellContent__text"
                        title="Job state"
                      >
                        Job state
                      </span>
                      <span
                        class="euiScreenReaderOnly"
                      >
                        Click to sort in ascending order
                      </span>
                    </span>
                  </button>
                </th>
                <th
                  aria-live="polite"
                  aria-sort="none"
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_rollup.rollup.continuous_4"
                  role="columnheader"
                  scope="col"
                >
                  <button
                    class="euiTableHeaderButton"
                    data-test-subj="tableHeaderSortButton"
                    type="button"
                  >
                    <span
                      class="euiTableCellContent"
                    >
                      <span
                        class="euiTableCellContent__text"
                        title="Continuous"
                      >
                        Continuous
                      </span>
                      <span
                        class="euiScreenReaderOnly"
                      >
                        Click to sort in ascending order
                      </span>
                    </span>
                  </button>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_metadata.rollup_metadata.continuous_5"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Next rollup window"
                    >
                      Next rollup window
                    </span>
                  </div>
                </th>
                <th
                  class="euiTableHeaderCell"
                  data-test-subj="tableHeaderCell_metadata_6"
                  role="columnheader"
                  scope="col"
                >
                  <div
                    class="euiTableCellContent"
                  >
                    <span
                      class="euiTableCellContent__text"
                      title="Rollup job status"
                    >
                      Rollup job status
                    </span>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                class="euiTableRow"
              >
                <td
                  class="euiTableRowCell euiTableRowCell--isMobileFullWidth"
                  colspan="8"
                >
                  <div
                    class="euiTableCellContent euiTableCellContent--alignCenter"
                  >
                    <span
                      class="euiTableCellContent__text"
                    >
                      <div
                        class="euiEmptyPrompt"
                        style="max-width: 45em;"
                      >
                        <span
                          class="euiTextColor euiTextColor--subdued"
                        >
                          <div
                            class="euiText euiText--medium"
                          >
                            <div
                              class="euiText euiText--medium"
                            >
                              <p>
                                Loading rollup jobs...
                              </p>
                            </div>
                          </div>
                        </span>
                      </div>
                    </span>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
